lib/rubocop/cop/rspec/leading_subject.rb

```diff
@@ -43,27 +45,25 @@ def on_block(node)
         def check_previous_nodes(node)
           node.parent.each_child_node do |sibling|
             if offending?(sibling)
-              add_offense(
-                node,
-                message: format(MSG, offending: sibling.method_name)
-              )
+              msg = format(MSG, offending: sibling.method_name)
+              add_offense(node, message: msg) do |corrector|
+                autocorrect(corrector, node)
+              end
             end
 
             break if offending?(sibling) || sibling.equal?(node)
           end
         end
 
-        def autocorrect(node)
-          lambda do |corrector|
-            first_node = find_first_offending_node(node)
-            RuboCop::RSpec::Corrector::MoveNode.new(
-              node, corrector, processed_source
-            ).move_before(first_node)
-          end
-        end
-
         private
 
+        def autocorrect(corrector, node)
+          first_node = find_first_offending_node(node)
```

Comment:
In a future refactoring, we can save the offending node in the check_previous_node method and just pass it to the autocorrect

Reply:
#960